vault/identity_store_entities.go

```diff
@@ -756,6 +780,33 @@ func (i *IdentityStore) mergeEntity(ctx context.Context, txn *memdb.Txn, toEntit
 			return errors.New("entity id to merge from does not belong to this namespace"), nil
 		}
 
+		// If we're not resolving a conflict, we check to see if
+		// any aliases conflict between the toEntity and this fromEntity:
+		if !forceMergeAliases && len(conflictingAliasIDsToKeep) == 0 {
```

Comment:
This for loop seemed to be a pre-validation for loop, as in, containing stuff like:
```
		if fromEntity == nil {
			return errors.New("entity id to merge from is invalid"), nil
		}
```

This for loop is definitely doing more validation than MFA and we need to do these checks in a 'validation' for loop before we do an actual merge. I don't really understand what makes this for loop sacred enough to warrant adding a third for loop - this for loop does required preparation pre-merge and while that does include MFA, it can include other stuff too as this is called in every case.

You're right that some checks (like the one I listed above) seem to be repeated in the second loop but that looks like a mistake as some of the code would never be hit.